src/main/scala/api/other/IcpBasedSurfaceFitting.scala

```diff
@@ -90,7 +90,18 @@ case class IcpBasedSurfaceFitting(model: StatisticalMeshModel, target: TriangleM
       }
 
       if (nbIterations > 0) {
-        recursion(newCoeff, nbIterations - 1, sigma, finalTrans)
+        try
+        {
+          recursion(newCoeff, nbIterations - 1, sigma, finalTrans)
+        }
+        catch
+        {
+          case e: Exception => {
+            println(s"Some error occured in IcpBasedSurfaceFITTING!!! Iteration: ${numIterations - nbIterations}) / ${numIterations}")
```

Comment:
Exceptions should be printed to stderr and not stdout. 
"An error" instead of "Some error".  
Capitalization of FITTING and !!! seems odd